### ISSUE TITLE
Feature to express the `row_group_size` as a string (eg. '250mb') in `write_parquet`

### ISSUE THREAD

**Issue by mkleinbort-ic** [pola-rs/polars#14069]
### Description

Most guidance on optimal parquet file layouts is expressed in terms of the size of row groups, not the number of rows.

Given polars has good knowledge of the data types of the columns, it should be able to figure out the row-number equivalent of "250mb".

So the ask is to allow for a string input to `write_parquet` that takes a string representing the desired row group size.

The [pl.DataFrame.estimated_size](https://docs.pola.rs/py-polars/html/reference/dataframe/api/polars.DataFrame.estimated_size.html) method already clarifies the accepted units:

`'b', 'kb', 'mb', 'gb', 'tb'`

So I propose the same be supported here.

Eg.

`'2000000b'`
`'2000kb'`
`'2mb'`
`'0.2gb'`

etc...

**Comment by deanm0000** [comment]
The `estimated_size` method is the memory footprint but parquets are compressed so it's not so easy to go from having an estimated_size of memory to the compressed size. That's especially true because there are a handful of compression algorithms to choose from, each of which have settable compression level settings.

Further still, the more row_groups there are, the worse effective compression is going to be.

**Comment by mkleinbort-ic** [comment]
Yes, I meant it as row-group size in terms of uncompressed data. It will not match the row group size on disk, but it's more practical than a row count.

In our pipelines we have tables with only two columns, and tables with each row is >1Mb.

Being able to express the target row group size at least in terms of the in-memory data schema would be helpful as a starting point for optimization.

**Comment by cjackal** [comment]
Just a pure curiosity - how do we efficiently estimate the rowcount given the max memory size for utf-8 columns? I can't come up with something as cheap as that for, e.g., FP32.
